src/server/lesson-controller.js

```diff
@@ -0,0 +1,49 @@
+import urls from '../utils/urls.js';
+import httpErrors from 'http-errors';
+import PageRenderer from './page-renderer.js';
+import { PAGE_NAME } from '../domain/page-name.js';
+import ServerConfig from '../bootstrap/server-config.js';
+import LessonService from '../services/lesson-service.js';
+import { validateParams } from '../domain/validation-middleware.js';
+import { getLessonParamsSchema } from '../domain/schemas/lesson-schemas.js';
+
+const { NotFound } = httpErrors;
+
+export default class LessonController {
+  static get inject() { return [ServerConfig, LessonService, PageRenderer]; }
+
+  constructor(serverConfig, lessonService, pageRenderer) {
+    this.serverConfig = serverConfig;
+    this.lessonService = lessonService;
+    this.pageRenderer = pageRenderer;
+  }
+
+  async handleGetLessonPage(req, res) {
+    const { lessonId } = req.params;
+    const routeWildcardValue = urls.trimSlashes(req.params[0]);
```

Comment:
I would suggest to use `removeLeadingSlash`, because the original slug could end with a trailing slash and then they wouldn't match anymore.

Reply:
Of course, we don't have nice sanitized slugs in the database. Changed, thanks!